### ISSUE TITLE
Problem with hibernate cache when generate the jhipster application

### ISSUE THREAD

**Issue by Joaggi** [sonalake/generator-jhipster-multitenancy#27]
Hello,

I use the jhipster-generator v4.13.1 in order to generate the main jhipster application. After that I use the command `npm install generator-jhipster-multitenancy` in order to install the multitenancy generator.

The problems occourse when I try to do `yo jhipster-multitenancy`, a prompt is given with `What is the alias given tenants in your application? (Company) ` after I press enter in this situation the next error occours

```
events.js:182
      throw er; // Unhandled 'error' event
      ^

ReferenceError: /home/joag/Proyectos/Munkys/prueba2/node_modules/generator-jhipster-multitenancy/generators/app/templates/src/main/java/package/domain/_User.java:26
    24| import com.fasterxml.jackson.annotation.JsonIgnore;
    25| import org.apache.commons.lang3.StringUtils;<% if (databaseType === 'sql') { %>
 >> 26| import org.hibernate.annotations.BatchSize;<% } %><% if (hibernateCache !== 'no' && databaseType === 'sql') { %>
    27| import org.hibernate.annotations.Cache;
    28| import org.hibernate.annotations.CacheConcurrencyStrategy;<% } %>
    29| import org.hibernate.validator.constraints.Email;

hibernateCache is not defined
    at eval (eval at compile (/home/joag/Proyectos/Munkys/prueba2/node_modules/ejs/lib/ejs.js:549:12), <anonymous>:28:8)
    at returnedFn (/home/joag/Proyectos/Munkys/prueba2/node_modules/ejs/lib/ejs.js:580:17)
    at Object.exports.render (/home/joag/Proyectos/Munkys/prueba2/node_modules/ejs/lib/ejs.js:384:37)
    at process (/home/joag/Proyectos/Munkys/prueba2/node_modules/mem-fs-editor/lib/actions/copy-tpl.js:12:18)
    at applyProcessingFunc (/home/joag/Proyectos/Munkys/prueba2/node_modules/mem-fs-editor/lib/actions/copy.js:13:16)
    at EditionInterface.exports._copySingle (/home/joag/Proyectos/Munkys/prueba2/node_modules/mem-fs-editor/lib/actions/copy.js:63:16)
    at files.forEach.file (/home/joag/Proyectos/Munkys/prueba2/node_modules/mem-fs-editor/lib/actions/copy.js:50:10)
    at Array.forEach (<anonymous>)
    at EditionInterface.exports.copy (/home/joag/Proyectos/Munkys/prueba2/node_modules/mem-fs-editor/lib/actions/copy.js:49:9)
    at EditionInterface.module.exports [as copyTpl] (/home/joag/Proyectos/Munkys/prueba2/node_modules/mem-fs-editor/lib/actions/copy-tpl.js:10:8)
```

I don't know if a I commit any mistake in the installation. The version is v0.9.5

**Comment by kalyandas75** [comment]
I got the same error

**Comment by Joaggi** [comment]
https://stackoverflow.com/questions/48044668/hibernate-cache-error-when-generate-the-jhipster-multitenancy-application-with-j

In StackOverflow Pascal Grimaud answer that the problem is the following

> It's because we change the configuration hibernateCache to cacheProvider and enableHibernateCache. You should open an issue in this project to inform the author, so maybe he can fix it and make a new release.

**Comment by ciarant** [comment]
Thanks for reporting the issue. We'll get on it this week.

**Comment by jduggan83** [comment]
Hi @Joaggi 
New version of the module v0.9.6 just published to npm. Closing this issue as this is now fixed.

**Comment by esulcz** [comment]
I've the same issue with JHipster Multitenancy Generator v0.9.6 and JHipster v4.14.0.
What could be the problem?

**Comment by ciarant** [comment]
Thanks @esulcz - we'll investigate.

**Comment by esulcz** [comment]
Additional information: It's working with brand new generated project, jhipster  version 4.14.4. But with existing project ( jhipster v4.14.0) i've got the following error:
Welcome to the JHipster Multitenancy Generator v0.9.6

? What is the alias given tenants in your application? Company
events.js:183
      throw er; // Unhandled 'error' event
      ^

ReferenceError: C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\generators\app\templates\src\main\java\package\domain\_User.java:93
    91|     import com.fasterxml.jackson.annotation.JsonIgnore;
    92|     import org.apache.commons.lang3.StringUtils;<% if (databaseType === 'sql') { %>
 >> 93|     import org.hibernate.annotations.BatchSize;<% } %><% if (enableHibernateCache) { %>
    94|     import org.hibernate.annotations.Cache;
    95|     import org.hibernate.annotations.CacheConcurrencyStrategy;<% } %>
    96|     import org.hibernate.validator.constraints.Email;

enableHibernateCache is not defined
    at eval (eval at compile (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\ejs\lib\ejs.js:549:12), <anonymous>:34:8)
    at returnedFn (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\ejs\lib\ejs.js:580:17)
    at Object.exports.render (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\ejs\lib\ejs.js:384:37)
    at process (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\mem-fs-editor\lib\actions\copy-tpl.js:12:18)
    at applyProcessingFunc (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\mem-fs-editor\lib\actions\copy.js:13:16)
    at EditionInterface.exports._copySingle (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\mem-fs-editor\lib\actions\copy.js:63:16)
    at files.forEach.file (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\mem-fs-editor\lib\actions\copy.js:50:10)
    at Array.forEach (<anonymous>)
    at EditionInterface.exports.copy (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\mem-fs-editor\lib\actions\copy.js:49:9)
    at EditionInterface.module.exports [as copyTpl] (C:\Users\esulcz\AppData\Roaming\npm\node_modules\generator-jhipster-multitenancy\node_modules\mem-fs-editor\lib\actions\copy-tpl.js:10:8)

**Comment by jduggan83** [comment]
@esulcz Thanks for the info on this. Im looking into this now. Our previous fix for this had intended to fix this issue from 4.13.1 and above. I will do some testing across versions and try and provide a fix soon.

**Comment by alexander-ozkan** [comment]
We have just released v1.0.0 of our module. This issue has been fixed for the latest version of JHipster -v5.1.0.
